src/utils/island.ts

```diff
@@ -0,0 +1,5 @@
+import type { H3Event } from 'h3'

```

Comment:
My personal preference would be to not use an extra file in  `/utils/`.
If this one-liner is going to be used only in one place, I'd rather see it directly  in the code.

Rationale for this: each time we add a file in `src/utils/`, we need to also augment the `unbuild` entry in package.json, which adds extra maintenance costs. Otherwise the build will fail. My opinion is that `src/utils/` should only contain functions that we need to import in more than one place, of which at least one belongs to the server runtime

Reply:
I think the extra entry would not be needed because the isIslandRequest utility is just internal and does not need to be exposed?

However, I agree that a new utility file for a simple function that is only used in one place is not needed. I will update the code and probably remove the check entirely and check for an existing nonce instead.

Reply:
Done.